modules/rest-util/src/blaze/middleware/fhir/output.clj

```diff
@@ -32,29 +32,37 @@
   (with-open [_ (prom/timer generate-duration-seconds "json")]
     (fhir-spec/unform-json body)))
 
-(defn- generate-xml** [body]
+(defn- xml-byte-array [body]
```

Comment:
`xml-byte-array` is, in many aspects, not a good name. First it doesn't convert some XML into a byte-array, second I don't thing the aspect that the return type is a byte-array has such an importance here, third, there is already an established name for generating an XML response body.

Also the convention to postfix function names with stars that need to be broken up in vertical terms, is quite idiomatic in Clojure. All the variants of `generate-xml` actually generate XML but the first one wraps metric collection and the second one wraps exception handling.

Reply:
I see your point. I've seen the star convention you mention in other portions of the code, it makes sense.

I only tried to make the name different in this case (my choice probably being not the best guess, as you explained above) because, within the whole `generate-xml[*,**,***]` function chain, this one was the only one that still used the body, and not the whole response. Having two functions whose name only differs by one star with two different interfaces seems error-prone, specially for someone who, in the not-so-near future comes and revisits that piece of code.

Or am I overthinking this?

Reply:
I see your point with the different interface of the functions. Weighting all options, I think, we should keep the original name.